src/test/java/uk/gov/companieshouse/api/accounts/validation/IntangibleAssetsValidatorTest.java

```diff
@@ -0,0 +1,174 @@
+
+        ReflectionTestUtils.setField(validator, UNEXPECTED_DATA_KEY, UNEXPECTED_DATA);
+
+        Errors errors = validator.validateIntangibleAssets(intangibleAssets, transaction, COMPANY_ACCOUNTS_ID, request);
+
+        assertEquals(1, errors.getErrorCount());
+        assertTrue(errors.containsError(createError(UNEXPECTED_DATA, "$.intangible_assets.goodwill.cost.at_period_start")));
+    }
+
+//    @Test
+//    @DisplayName("First year filer - provides only cost fields in sub resource")
+//    void firstYearFilerProvidesOnlyCostFieldsInSubResource() throws ServiceException, DataException {
+//
+//        when(companyService.isMultipleYearFiler(any(Transaction.class))).thenReturn(false);
+//
+//        IntangibleAssetsResource goodwill = new IntangibleAssetsResource();
+//
+//        Cost cost = new Cost();
+//        cost.setAdditions(1L);
+//
+//        goodwill.setCost(cost);
+//
+//        IntangibleAssets intangibleAssets = new IntangibleAssets();
+//        intangibleAssets.setGoodwill(goodwill);
+//
+//        ReflectionTestUtils.setField(validator, INCORRECT_TOTAL_KEY, INCORRECT_TOTAL);
+//
+//        Errors errors = validator.validateIntangibleAssets(intangibleAssets, transaction, COMPANY_ACCOUNTS_ID, request);
+//
+//        assertEquals(1, errors.getErrorCount());
+//        assertTrue(errors.containsError(createError(INCORRECT_TOTAL, "$.intangible_assets.goodwill.cost.at_period_end")));
+//    }
+
+    @Test
+    @DisplayName("Single year filer - cost fields don't total in sub resource")
+    void singleYearFilerCostFieldsDoNotTotalInSubResource() throws ServiceException, DataException {
+
+        when(companyService.isMultipleYearFiler(any(Transaction.class))).thenReturn(false);
+
+        IntangibleAssetsResource goodwill = new IntangibleAssetsResource();
+
+        Cost cost = new Cost();
+        cost.setAdditions(1L);
+        cost.setAtPeriodEnd(2L);
+        goodwill.setCost(cost);
+        goodwill.setNetBookValueAtEndOfCurrentPeriod(2L);
+
+        IntangibleAssets intangibleAssets = new IntangibleAssets();
+        intangibleAssets.setGoodwill(goodwill);
+
+        ReflectionTestUtils.setField(validator, INCORRECT_TOTAL_KEY, INCORRECT_TOTAL);
+
+        Errors errors = validator.validateIntangibleAssets(intangibleAssets, transaction, COMPANY_ACCOUNTS_ID, request);
+
+        assertEquals(1, errors.getErrorCount());
+        assertTrue(errors.containsError(createError(INCORRECT_TOTAL, "$.intangible_assets.goodwill.cost.at_period_end")));
+    }
+
+    @Test
+    @DisplayName("First year filer - Does not provide cost at period end in sub resource")
+    void firstYearFilerDoesNotProvideCostAtPeriodEndInSubResource() throws ServiceException, DataException {
+
+        when(companyService.isMultipleYearFiler(any(Transaction.class))).thenReturn(false);
+
+        IntangibleAssetsResource goodwill = new IntangibleAssetsResource();
+
+        Cost cost = new Cost();
+
+        goodwill.setCost(cost);
```

Comment:
Remove the variable assignment at line 153 and just set: `goodwill.setCost(new Cost());`